mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.h

```diff
@@ -62,6 +62,8 @@ struct ContractionDimensions {
 /// `k`, indices are returned in sorted order.
 /// Returns a failure if any of `m`, `n` or `k` is empty.
 FailureOr<ContractionDimensions> inferContractionDims(LinalgOp linalgOp);
+FailureOr<ContractionDimensions>
```

Comment:
Out of curiosity, why is this something useful? What use case do you have in mind?

Reply:
I need it here: https://github.com/openxla/iree/pull/15984
TLDR, We have an attribute on types tracking the indexing maps, so we don't have access to the actual op.

But this can also be useful more generally in cases where you only have access to indexing maps. For example, doing indexing map analysis before creating a new op. It seems like a potentially useful util to have to me.

Reply:
Ok, thanks. I don't have a strong opinion here, but I would expect something more generic if we go down to affine maps (i.e., it is unclear why there is a restriction on the number of affine maps).

Reply:
Given the definition of the ContractionOpInterface, I think it makes sense to keep it restricted to 3 maps (lhs, rhs, out) https://github.com/llvm/llvm-project/blob/74a09bd1ec6066d56880df1ae1a2c0258442cee9/mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td#L19-L35
Is there a more general form of a tensor contraction that you are thinking of? Perhaps something like that could live elsewhere to avoid confusion with the `ContractionOpInterface` defined here.

Reply:
Sorry for the latency, I lost this PR. Yes, you are right `ContractionOpInterface` already assumes two inputs and one output. It makes sense to restrict to 3 maps only then.